torchmetrics/text/meteor.py

```diff
@@ -0,0 +1,134 @@
+
+import warnings
+from typing import Any, Callable, List, Literal, Optional, Tuple, Union
+
+from torch import Tensor
+
+from torchmetrics import Metric
+from torchmetrics.functional.text.meteor import (
+    _meteor_score_compute,
+    _meteor_score_update,
+    _METEORScoreComponents,
+    _NLTKStemmerWrapper,
+    _NLTKWordnetWrapper,
+)
+
+
+class METEORScore(Metric):
+    """Calculate `BLEU score`_ of machine translated text with one or more references.
+
+    Args:
+        stemmer:
+            A name of stemmer from `nltk` package to be used.
+        wordnet:
+            A name of wordnet corpus from `nltk` package to be used.
+        alpha:
+            A parameter for controlling relative weights of precision and recall.
+        beta:
+            A parameter for controlling shape of penalty as a function of as a function of fragmentation.
+        gamma:
+            A relative weight assigned to fragmentation penalty.
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+
+    Example:
+        >>> predictions = ['the cat is on the mat']
+        >>> references = [['there is a cat on the mat']]
+        >>> meteor_score = METEORScore()
+        >>> meteor_score.update(references, predictions)
+        >>> meteor_score.compute()
+        tensor([0.6464])
+
+    References:
+    """
+
+    is_differentiable = False
+    higher_is_better = True
+    meteor_score_components: List[Tuple[_METEORScoreComponents]]
+
+    def __init__(
+        self,
+        stemmer: Literal["porter"] = "porter",
+        wordnet: Literal["wordnet"] = "wordnet",
+        alpha: float = 0.9,
+        beta: float = 3.0,
+        gamma: float = 0.5,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Optional[Callable] = None,
+    ):
+        warnings.warn(
+            "Current implementation follows the original METEOR metric and thus is not suitable for reporting results "
+            "in research papers."
+        )
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        self.stemmer = _NLTKStemmerWrapper(stemmer)
+        self.wordnet = _NLTKWordnetWrapper(wordnet)
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
```

Comment:
I assume we need to do some validation of these parameters

Reply:
Yes, `gamma` should be also between 0 and 1. Not sure about `beta`. I'll have a look :]